module/executiondatasync/optimistic_sync/execution_result/info_provider.go

```diff
@@ -53,39 +53,46 @@ func NewExecutionResultInfoProvider(
 // based on the provided criteria.
 //
 // Expected errors during normal operations:
-//   - [common.InsufficientExecutionReceipts]: Found insufficient receipts for given block ID.
-//   - [storage.ErrNotFound]: If the data was not found.
+//   - [storage.ErrNotFound]: If the execution receipts for the block ID are not found.
+//   - [optimistic_sync.ErrBlockBeforeNodeHistory]: If the request is for the spork root block, and the node was bootstrapped
+//     from a newer block.
+//   - [optimistic_sync.ErrForkAbandoned]: If the execution fork of an execution node from which we were getting the
+//     execution results was abandoned.
+//   - [optimistic_sync.ErrNotEnoughAgreeingExecutors]: If there are not enough execution nodes that produced the
+//     execution result.
+//   - [optimistic_sync.ErrRequiredExecutorNotFound]: If the criteria's required executor is not in the group of
+//     execution nodes that produced the execution result.
```

Comment:
these errors relate to a specific execution result, but this method queries a set of results for a block. Instead of returning these, I think this method should simply state that no result matched the requested criteria.

looking at the returned errors, it seems there are a few cases:
* The block has not been executed yet (no receipts found)
* The block is before the node's available history
* No results found that matched the criteria (unsealed block)
* No results found that matched the criteria (sealed block)
* The criteria was invalid